File: 백준/1번문제/S0okJu/Q2217.py
"""

1. 아이디어 
w(물체의 중량) / k(로프의 개수) = 로프의 중량 
우리가 구하고자 하는 것은 최대 물체의 중량 
문제에서는 로프의 최대 중량이 주어짐 
- w(물체의 중량) max / k(로프의 개수) min = 로프의 중량 max
- max 물체의 중량 = 로프의 중량 min * 로프의 개수 max

2. 시간복잡도 :
O(n)

3. 자료구조 : x

"""
def solution(n, max_w):
    sorted_max_w = sorted(max_w)
    k = n 
    hw = 0
    for j in range(len(sorted_max_w)):
        w = sorted_max_w[j] * (k-j)   
        hw = max(hw,w)
    
    print(hw)
    

if __name__ == '__main__':
    n = int(input())
    max_w = list()
    for i in range(n):
        w = int(input())
        max_w.append(w)
    
    solution(n, max_w)

File: 백준/1번문제/최원준/Q2217.java
// https://www.acmicpc.net/problem/
package 백준;

/*


1. 아이디어 :
    밧줄을 기준으로 내림차순.
    각 밧줄마다 그 전 밧줄까지 포함하여 평균을 계산
2. 시간복잡도 :
    O( nlogn )
3. 자료구조 :
    배열

 */
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Collections;

class Q2217 {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    public static void main(String[] args) throws IOException {
        int n = Integer.parseInt(br.readLine());

        ArrayList<Integer> ropes = new ArrayList<>();
        for (int i=0; i<n; i++) {
            ropes.add(Integer.parseInt(br.readLine()));
        }

        Collections.sort(ropes, (a,b) -> {
            return b-a;
        });

        int ans = 0;
        for (int i=0; i<n; i++) {
            ans = Math.max(ans, ropes.get(i) * (i+1));
        }
        System.out.println(ans);
    }
}

/*
import sys
input = lambda: sys.stdin.readline().rstrip()

def solution(n, ropes):
    return max([ropes[i] * (i+1) for i in range(len(ropes))])

if __name__ == '__main__':
    n = int(input())
    ropes = sorted([int(input()) for _ in range(n)], key=lambda x: -x)
    print(solution(n, ropes))
 */
